lib/ClangImporter/ImportDecl.cpp

```diff
@@ -1500,6 +1500,12 @@ namespace {
 
       Type SwiftType;
       if (Decl->getDeclContext()->getRedeclContext()->isTranslationUnit()) {
+        // Ignore the 'id' typedef. We want to bridge the underlying
+        // ObjCId type.
+        if (Name.str() == "id" && Impl.SwiftContext.LangOpts.EnableIdAsAny) {
```

Comment:
From what I've observed, we *do* recognize it from the table in `BridgedTypes.def` and immediately map it to `AnyObject` without this early exit.